qemu/tests/hugepage_specify_node.py

```diff
@@ -0,0 +1,81 @@
+import logging
+
+from avocado.utils import memory
+
+from virttest import env_process
+from virttest import error_context
+from virttest import utils_misc
+
+from virttest.utils_numeric import normalize_data_size
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Qemu allocate hugepage from specify node.
+    Steps:
+    1) Setup total of 4G mem hugepages for specify node.
+    2) Setup total of 1G mem hugepages for idle node.
+    3) Mount this hugepage to /mnt/kvm_hugepage.
+    4) Boot guest only allocate hugepage from specify node.
+    5) Check the hugepage used from every node.
+    :params test: QEMU test object.
+    :params params: Dictionary with the test parameters.
+    :params env: Dictionary with test environment.
+    """
+    hugepage_size = memory.get_huge_page_size()
+    mem_size = int(normalize_data_size("%sM" % params["mem"], "K"))
+    idle_node_mem = int(normalize_data_size("%sM" % params["idle_node_mem"], "K"))
+
+    error_context.context("Get host numa topological structure.", logging.info)
+    host_numa_node = utils_misc.NumaInfo()
+    node_list = host_numa_node.online_nodes_withcpumem
```

Comment:
node_list = host_numa_node.online_nodes_withmem

Reply:
So if a node has mem, but the cpu is offline,  it not affect the test?

Reply:
No, it wouldn't matter in this case.

Reply:
ok, corrected.